### PR TITLE
docs: add doc about running docs locally

### PR DESCRIPTION
**Pull request by kapunahelewong**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Adds step-by-step instructions on running docs locally

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

The location in the nav is not final. It will depend on the forthcoming Contributing section.
Converting to draft until Contributing section is ready.

### CODE REVIEW

File: aio/content/guide/docs-running-locally.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+## How to amend a commit
+
+Amend a commit when you want to change the commit message.
+
+1. On your **feature branch**: `git commit --amend`
+1. Your default editor opens.
+  Change your commit message.
+  Save and close.
+1. Back at the command line, run `git log` or `git log --oneline` to confirm that your commit message is as you intend.
+1. Push up to GitHub.
```
Comment (BioPhoton) [inline aio/content/guide/docs-running-locally.md:214]: Maybe add `git push` again?

**Review by BioPhoton — COMMENTED** [review]
Thanks for this more than helpful document!
I believe this will also help people outside ou angular!
<3

File: aio/content/guide/docs-running-locally.md (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/guide/docs-running-locally.md:214]: Thanks for your suggestions, @BioPhoton. I've added them all. 🌟

**Review by IgorMinar — CHANGES_REQUESTED** [review]
thanks @kapunahelewong! however the navbar change doesn't look right:

Screen Shot 2020-06-10 at 1 48 26 AM

shouldn't this guide be hidden from normal navigation?

And as for the content, @gkalpak can you please help with the review?

**Review by gkalpak — CHANGES_REQUESTED** [review]
I like where this is heading. Thx, @kapunahelewong :tada:

**Comment by kapunahelewong** [comment]
> thanks @kapunahelewong! however the navbar change doesn't look right:
> 
> Screen Shot 2020-06-10 at 1 48 26 AM
> 
> shouldn't this guide be hidden from normal navigation?
> 
> And as for the content, @gkalpak can you please help with the review?

Thanks, @IgorMinar! @aikidave and I are working on docs to help contributors, and I believe the intention is for there to be a section in the nav for it, but not how I've unceremoniously placed it! It _does_ look odd like that. 😅

File: aio/content/guide/docs-running-locally.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+The following sections show you how to fork the Angular repo in GitHub, get a local copy of you can work on, and how to run the docs locally.
+
+{@a github}
+
+### In GitHub
+
+The two preliminary steps in this section are in GitHub.
+After completing these two steps, continue on to the following section which shows you how to use the terminal to complete your local Angular setup.
+
+1. [Fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) [angular/angular](https://github.com/angular/angular).
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/docs-running-locally.md:25]: Oh that's good. Thanks!

File: aio/content/guide/docs-running-locally.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+<code-example language="sh" class="code-shell">
+
+ docs: fix typo in template syntax
+
+ Fixes ####
+
+</code-example>
+
+Replace `####` with the issue number you're fixing.
+Alternatively, you can use the word `Closes`.
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/docs-running-locally.md:167]: Linked instead to github docs.Thanks!

File: aio/content/guide/docs-running-locally.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+1. Push up to GitHub with `git push`.
+  Your PR should reflect the updated commit message.
+  You might need `-f` to force the push.
+  When force pushing, be sure you understand what it entails before you do it.
+  If you're collaborating with someone else, coordinate any force pushing as it overwrites anything else on that branch in GitHub.
+
+## Clearing out commits that aren't yours in your PR
+
+Sometimes when you merge, as in `git pull`,  `master` into your feature branch, you can find a number of commits that aren't yours in your PR.
+To avoid this, consider using `rebase` instead of `merge`.
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/docs-running-locally.md:225]: > Why would you "git pull master into your feature branch"?

You wouldn't, but I think this is how many people end up with those extraneous commits in their PR's. You make a good point though, we shouldn't put how to make the mistake in the documentation, just the fix. I'll remove it.

File: aio/content/guide/docs-running-locally.md [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+
+Before doing this, since your branch is in an undesirable state, back up your work.
+Copy and paste it somewhere outside of your project so that when `git` does its work it doesn't overwrite your back up.
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/guide/docs-running-locally.md:228]: Yes, that was my thinking too. If someone is unsure when trying to fix their git problem, knowing that they have a back up is reassuring.

File: aio/content/guide/docs-running-locally.md [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+
+## Prerequisites
+
+* You need a [GitHub](https://github.com/join) account and a basic [knowledge of git](https://guides.github.com/introduction/git-handbook/).
+  If you use Windows, you may find [Git for Windows](https://gitforwindows.org/)
+ helpful.
+* Make sure you've read the Angular.io [CONTRIBUTING.md](https://github.com/angular/angular/blob/master/CONTRIBUTING.md).
+* Familiarity with the [Angular documentation style guide](guide/docs-style-guide).
+
+<hr />
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:14]: We can remove the <hr /> tags. I think the latest styles update makes them unnecessary?

File: aio/content/guide/docs-running-locally.md [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+
+## Getting your own copy of the repo
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:16]: In our new contributor's guide, we point users to the CONTRIBUTING.md file for this information. Do you think that's sufficient?

If you think this content is important to keep, let's create a separate topic for this information. That way, we can link to it from other topics as well!

File: aio/content/guide/docs-running-locally.md [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+
+The following sections show you how to fork the Angular repo in GitHub, get a local copy  you can work on, and how to build and run the docs app locally.
+
+{@a github}
+
+### In GitHub
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:22]:
```suggestion
### Forking a repository using the GitHub UI
```

File: aio/content/guide/docs-running-locally.md [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+
+1. [Fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) the[angular/angular](https://github.com/angular/angular).
+1. In your new fork, click the "Clone or Download" button to expand the Clone functionality and click the copy-to-clipboard icon to copy the URL of your fork.
+
+<div class="lightbox">
+  <img src="generated/images/guide/docs-running-locally/how-to-clone-repo.png"
+    alt="screenshot of clone button on GitHub">
+</div>
+
+### In terminal
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:35]:
```suggestion
### Building the documentation locally
```

File: aio/content/guide/docs-running-locally.md [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+
+Switch to your terminal and take the following steps:
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:37]:
```suggestion
After you've created a local copy of your forked repository, you can build the Angular documentation. 
```

File: aio/content/guide/docs-running-locally.md [hunk 11]
```diff
@@ -0,0 +1,306 @@
...
+
+1. Clone your fork: `git clone the-angular-repo-you-want-to-clone`, where `the-angular-repo-you-want-to-clone` is the link you copied in GitHub's ["Clone or Download" dialogue](guide/docs-running-locally#github).
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:39]: Line 39 isn't necessary.

File: aio/content/guide/docs-running-locally.md [hunk 12]
```diff
@@ -0,0 +1,306 @@
...
+1. Go into the `angular` directory: `cd angular`
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:40]:
```suggestion
1. Navigate to the `angular` directory: `cd angular`
```

File: aio/content/guide/docs-running-locally.md [hunk 13]
```diff
@@ -0,0 +1,306 @@
...
+1. Checkout a new branch where you'll do your work: `git checkout -b yourname-branch-topic`
+1. Go into the `aio` directory: `cd aio`
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:42]:
```suggestion
1. Navigate to the `aio` directory: `cd aio`
```

File: aio/content/guide/docs-running-locally.md [hunk 14]
```diff
@@ -0,0 +1,306 @@
...
+1. Setup `aio` with `yarn`: `yarn setup`
+1. Start the app and watch for changes: `yarn serve-and-sync`
+1. Open a second tab or window so you can issue git commands as you work.
+
+<div class="alert is-helpful">
+
+As an alternative to running `yarn serve-and-sync`, you can instead run `yarn start` and `yarn docs-watch` in two separate tabs.
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:49]:
```suggestion
As an alternative to running `yarn serve-and-sync`, you can instead run `yarn start` and `yarn docs-watch` in two separate terminals.
```

File: aio/content/guide/docs-running-locally.md [hunk 15]
```diff
@@ -0,0 +1,306 @@
...
+
+</div>
+
+Note that if you already have yarn running and you switch branches, the yarn processes might break.
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:53]:
```suggestion
If you already have yarn running and you switch branches, the yarn processes might break.
```

Should this be a warning?

File: aio/content/guide/docs-running-locally.md [hunk 16]
```diff
@@ -0,0 +1,306 @@
...
+In this case, `Ctrl`/`Cmd`+`C` and rerun the `yarn` command(s).
+
+### In the browser
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:56]:
```suggestion
### Viewing the documentation
```

File: aio/content/guide/docs-running-locally.md [hunk 17]
```diff
@@ -0,0 +1,306 @@
...
+
+If the above steps were successful, you should be able to view your local copy of the docs in your browser.
+
+1. Go to [localhost:4200](http://localhost:4200).
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:60]: A single numbered list seems odd. Maybe just:

To view the documentation, open a browser window and navigate to localhost:4200.

File: aio/content/guide/docs-running-locally.md [hunk 18]
```diff
@@ -0,0 +1,306 @@
...
+
+## Where the docs are
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:62]:
```suggestion
## Finding the source files for documentation
```

File: aio/content/guide/docs-running-locally.md [hunk 19]
```diff
@@ -0,0 +1,306 @@
...
+
+Many of the docs on [angular.io](https://www.angular.io) have an _Edit_, and often _View source_, icon at the top right, which opens the file on GitHub.
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:64]:
```suggestion
Sometimes, it's difficult to find the source file of the topic you want to update. Many of the docs on [angular.io](https://www.angular.io) have an _Edit_, and often _View source_, icon at the top right, which opens the file on GitHub.
```

File: aio/content/guide/docs-running-locally.md [hunk 20]
```diff
@@ -0,0 +1,306 @@
...
+    <td><code>aio/content/examples</code></td>
+  </tr>
+  <tr>
+    <td>API docs</td>
+    <td>Go to the API doc in question on angular.io and look near the top in the table to find the source. That's the file you edit in the API docs.</td>
+  </tr>
+
+</table>
+
+## Getting your work on GitHub
```
Comment (aikidave) [inline aio/content/guide/docs-running-locally.md:100]: Maybe just point to CONTRIBUTING.md?

**Review by aikithoughts — COMMENTED** [review]
A good start. I question whether we need all the information on using Git? Seems like we could point to the CONTRIBUTING guide, which covers this information already.

If you think we should have this content, let's discuss creating a topic or two that are specific to using Git. That would allow this topic to remain more focused, and it would allow us to more easily link to git-related topics from other places in the docs!

**Comment by jessicajaniuk** [comment]
@aikidave We're going to close this, but feel free to re-open / move the contents over to a new PR if this is still necessary.

**Pull request closed without merging**